src/inmanta/env.py

```diff
@@ -1145,8 +1145,7 @@ def init_env(self) -> None:
                 # On linux based systems, the python version is in the path to the site packages dir:
                 if not os.path.exists(self.site_packages_dir):
                     raise VenvActivationFailedError(
-                        msg=f"Unable to use virtualenv at {self.env_path} because its Python version "
-                        "is different from the Python version of this process."
+                        msg=f"Unable to use virtualenv at {self.env_path} as {self.site_packages_dir} does not exist"
```

Comment:
Do you mean something like this?

```
if os.path.isdir(self.env_path) and os.listdir(self.env_path):
    # Make sure the venv hosts the same python version as the running process
    if sys.platform.startswith("linux"):
        # Check if the python binary exists in the environment's bin directory
        python_binary = os.path.join(self.env_path, "bin", "python")
        if not os.path.exists(python_binary):
            raise VenvActivationFailedError(
                msg=f"Unable to use virtualenv at {self.env_path} as the directory is not empty but {self.site_packages_dir} does not exist"")
        # On linux based systems, the python version is in the path to the site packages dir:
        if not os.path.exists(self.site_packages_dir):
            raise VenvActivationFailedError(
                msg=f"Unable to use virtualenv at {self.env_path} because its Python version "
                        "is different from the Python version of this process.")
```

Reply:
And maybe use `self.python_path` instead of the hardcoded path?

Reply:
I think the change is done :)